orbit/pkg/platform/platform_windows.go

```diff
@@ -176,3 +178,124 @@ func GetProcessByName(name string) (*gopsutil_process.Process, error) {
+// library. This package smbios provides detection and access to System Management BIOS (SMBIOS) and
+// Desktop Management Interface (DMI) data and structures across: https://github.com/digitalocean/go-smbios
+// This function should work as is on Linux thanks to the go-smbios interface abstraction. See the
+// list of supported OSes on the go-smbios documentation.
+// The windows go-smbios implementations calls to GetSystemFirmwareTable()
+func hardwareGetSMBiosUUID() (string, error) {
+	// SMBIOS data in operating system-specific location
+	streamReader, smBIOSRawData, err := smbios.Stream()
+	if err != nil {
+		return "", fmt.Errorf("failed to open stream: %v", err)
+	}
+
+	// Ensure that stream will be closed
+	defer streamReader.Close()
+
+	// Decode SMBIOS structures from the stream.
+	decoder := smbios.NewDecoder(streamReader)
+	structSMBIOSdata, err := decoder.Decode()
+	if err != nil {
+		return "", fmt.Errorf("failed to decode BIOS structures: %v", err)
+	}
+
+	// Determine SMBIOS version and table location from entry point
+	biosMajor, biosMinor, _ := smBIOSRawData.Version()
+
+	// SMBIOS constants from spec here - https://www.dmtf.org/sites/default/files/standards/documents/DSP0134_3.1.1.pdf
+	const systemInformationType uint8 = 0x01 // System Information indicator
+	const minBiosStructSize uint8 = 0x1b     // Section 7.2 on the SMBIOS specification (0x1a min header size + null character)
+	const uuidOffset uint8 = 0x4             // UUID offset in System Information (Type 1) structure
+	const revMajorVersion int = 0x3          // SMBIOS revision that most of the current BIOS have - v3 specs were released in 2015
+	const minLegacyMajorVersion int = 0x2    // Minimum SMBIOS Major rev that supports UUID little-endian encoding
+	const minLegacyMinorVersion int = 0x6    // Minimum SMBIOS Minor rev that supports UUID little-endian encoding
+	const uuidSize int = 0x10                // UUID size is calculated with field offset value (0xA) + node field length (6 bytes)
+
+	var smBiosUUID string = ""
+
+	// Walking the obtained SMBIOS data
+	for _, rawBiosStruct := range structSMBIOSdata {
+		if (rawBiosStruct.Header.Type == systemInformationType) && (rawBiosStruct.Header.Length >= minBiosStructSize) {
+			uuidBytes := rawBiosStruct.Formatted[uuidOffset:]
+
+			// UUID field sanity check for null values
+			// Logic is based on https://github.com/ContinuumLLC/godep-go-smbios/blob/ab7c733f1be8e55ed3e0587d1aa2d5883fe8801e/smbios/decoder.go#L135
+			only0xFF, only0x00 := true, true
+			for i := 0; i < uuidSize && (only0x00 || only0xFF); i++ {
+				if uuidBytes[i] != 0x00 {
+					only0x00 = false
+				}
+				if uuidBytes[i] != 0xFF {
+					only0xFF = false
+				}
+			}
+
+			if only0xFF {
+				return "", errors.New("UUID is not currently present in the system, but it can be set.")
+			}
+			if only0x00 {
+				return "", errors.New("UUID is not present in the system.")
```

Comment:
Is it ever possible for there to be multiple entries, one where `only0xFF` or `only0x00` are true, and another that has the actual value? Just wondering whether this could possibly be a premature error return.

Reply:
The logic will check only one UUID entry at a time (16 bytes). I've added some unit tests to ensure that logic behaves as expected against different inputs.